Formula/libgcrypt.rb

```diff
@@ -41,6 +41,12 @@ def install
       MachO::Tools.change_install_name("#{buildpath}/tests/.libs/random",
                                        "#{lib}/libgcrypt.20.dylib",
                                        "#{buildpath}/src/.libs/libgcrypt.20.dylib")
+
+      # The above operation modified tests/.libs/random and
+      # broke the validity of its code signature. Have to
+      # re-sign the binary so that it can be authorized to
+      # run on Apple Silicon.
+      system("codesign", "-s", "-", "#{buildpath}/tests/.libs/random")
```

Comment:
Should this be run only on ARM? I think `Hardware::CPU.arm?` may work.

Cf. https://github.com/Homebrew/brew/pull/9102